src/Microsoft.ML.Transforms/Text/NgramHashingTransformer.cs

```diff
@@ -288,506 +269,471 @@ public ColInfoEx(ModelLoadContext ctx)
-                _bindings.Types[iinfo] = new VectorType(NumberType.Float, 1 << _exes[iinfo].HashBits);
-        }
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(args, nameof(args));
+            env.CheckValue(input, nameof(input));
 
-        private static int GetAndVerifyInvertHashMaxCount(Arguments args, Column col, ColInfoEx ex)
-        {
-            var invertHashMaxCount = col.InvertHash ?? args.InvertHash;
-            if (invertHashMaxCount != 0)
+            env.CheckValue(args.Column, nameof(args.Column));
+            var cols = new ColumnInfo[args.Column.Length];
+            using (var ch = env.Start("ValidateArgs"))
             {
-                if (invertHashMaxCount == -1)
-                    invertHashMaxCount = int.MaxValue;
-                Contracts.CheckUserArg(invertHashMaxCount > 0, nameof(args.InvertHash), "Value too small, must be -1 or larger");
-                // If the bits is 31 or higher, we can't declare a KeyValues of the appropriate length,
-                // this requiring a VBuffer of length 1u << 31 which exceeds int.MaxValue.
-                if (ex.HashBits >= 31)
-                    throw Contracts.ExceptUserArg(nameof(args.InvertHash), "Cannot support invertHash for a {0} bit hash. 30 is the maximum possible.", ex.HashBits);
+
+                for (int i = 0; i < cols.Length; i++)
+                {
+                    var item = args.Column[i];
+                    cols[i] = new ColumnInfo(item.Source ?? new string[] { item.Name },
+                        item.Name,
+                        item.NgramLength ?? args.NgramLength,
+                        item.SkipLength ?? args.SkipLength,
+                        item.AllLengths ?? args.AllLengths,
+                        item.HashBits ?? args.HashBits,
+                        item.Seed ?? args.Seed,
+                        item.Ordered ?? args.Ordered,
+                        item.InvertHash ?? args.InvertHash,
+                        item.RehashUnigrams ?? args.RehashUnigrams
+                        );
+                };
             }
-            return invertHashMaxCount;
+            return new NgramHashingTransformer(env, input, cols).MakeDataTransform(input);
         }
 
-        private void GetTerms(int iinfo, ref VBuffer<ReadOnlyMemory<char>> dst)
+        // Factory method for SignatureLoadModel.
+        private static NgramHashingTransformer Create(IHostEnvironment env, ModelLoadContext ctx)
         {
-            Host.Assert(0 <= iinfo && iinfo < _exes.Length);
-            Host.Assert(_slotNames[iinfo].Length > 0);
-            _slotNames[iinfo].CopyTo(ref dst);
+            Contracts.CheckValue(env, nameof(env));
+            var host = env.Register(nameof(NgramHashingTransformer));
+
+            host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel(GetVersionInfo());
+            return new NgramHashingTransformer(host, ctx);
         }
```

Comment:
I think you are repeating some checks here and in NgramHashingTransformer(IHostEnvironment env, ModelLoadContext ctx). Maybe just have one method, the constructor?